lib/stinger_core/src/stinger.c

```diff
@@ -972,85 +992,106 @@ etype_begin (stinger_t * S, int64_t v, int etype)
 
 MTA ("mta inline")
 void
-update_edge_data (struct stinger * S, struct stinger_eb *eb,
-                  uint64_t index, int64_t neighbor, int64_t weight,
-		  int64_t ts, int64_t operation)
+update_edge_data_and_direction (struct stinger * S, struct stinger_eb *eb,
+                  uint64_t index, int64_t neighbor, int64_t in_weight,
+                  int64_t ts, int64_t direction, int64_t operation)
 {
   struct stinger_edge * e = eb->edges + index;
 
   /* insertion */
   if (neighbor >= 0) {
-    switch (operation) {
-      case EDGE_WEIGHT_SET:
-	e->weight = weight;
-	break;
-      case EDGE_WEIGHT_INCR:
-	stinger_int64_fetch_add(&(e->weight), weight);
-	break;
+    int64_t weight = readfe (&(e->weight));
+    
+    if (direction & STINGER_EDGE_DIRECTION_OUT) {
+      if (operation & EDGE_WEIGHT_SET) {
+        //e->weight = weight;
```

Comment:
Why is the update of e->weight deferred until later? Anyways, commented out code should not be checked in.